elephant/conversion.py

```diff
@@ -564,6 +564,8 @@ def _check_consistency(self, spiketrains, binsize, num_bins, t_start,
             raise TypeError(
                 "Number of bins (num_bins) is not an integer or < 0: " + str(
                     num_bins))
+        elif binsize.units != self.t_start.units:
```

Comment:
These two lines only handle the inconsistency of t_start.units and binsize.units and disregards the case where t_start.units != t_stop.units.

To fix this, these two lines can be deleted and instead the calculation in the bin_edges() function should be changed. See below

Reply:
Change lines 586-589 to:

       return pq.Quantity(np.linspace(self.t_start.rescale('s').magnitude,
                                       self.t_stop.rescale('s').magnitude,
                                       self.num_bins + 1, endpoint=True), units='s').rescale(self.binsize.units)

Reply:
@rgutzen thanks for the comments. Wouldn't it make sense to have the same units for `start`, `stop` and `binsize`? Otherwise only the `bin_edges` property would be rescaled to seconds. I coulnd't find any side-effect so far, but they could appear with different units.

Reply:
@alperyeg the idea was that the units for each variable remain the same as entered by the user. Rescaling them might lead to confusion (e.g. if only `.magnitude` is used).
The `bin_edges` then take on the units of `binsize`, only within the calculation in between the variables are all rescaled to seconds.